From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Wed, 22 May 2024 13:27:35 +0100
Subject: [PATCH] Limit network_security_rule *_application_security_group_ids
 to 1

This should likely be applied using MaxItemsOne in the bridge.

diff --git a/internal/services/network/network_security_rule_resource.go b/internal/services/network/network_security_rule_resource.go
index 800adca50c..c53cda7218 100644
--- a/internal/services/network/network_security_rule_resource.go
+++ b/internal/services/network/network_security_rule_resource.go
@@ -136,7 +136,7 @@ func resourceNetworkSecurityRule() *pluginsdk.Resource {
 			//lintignore:S018
 			"source_application_security_group_ids": {
 				Type:     pluginsdk.TypeSet,
-				MaxItems: 10,
+				MaxItems: 1,
 				Optional: true,
 				Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
 				Set:      pluginsdk.HashString,
@@ -145,7 +145,7 @@ func resourceNetworkSecurityRule() *pluginsdk.Resource {
 			//lintignore:S018
 			"destination_application_security_group_ids": {
 				Type:     pluginsdk.TypeSet,
-				MaxItems: 10,
+				MaxItems: 1,
 				Optional: true,
 				Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
 				Set:      pluginsdk.HashString,
